### DISCUSSION TITLE
[feature] support workers

### DISCUSSION THREAD

**Discussion by Salakar** [zendive/jsdiff#27] [General]
Thanks for this extension, do you have any plans to support workers, e.g. selecting a worker in chrome and console.diff does not exist currently and is only available on 'top':

![image](https://github.com/zendive/jsdiff/assets/5347038/07e3b02d-36d7-498c-8eb3-8b2bf1630767)
![image](https://github.com/zendive/jsdiff/assets/5347038/966c98b9-18d4-4e20-9bee-cd208f624573)

**Comment by zendive** [comment]
@Salakar appreciate. There is no extension API for injecting content scripts into page workers. Only to tab's page and its iframes. Also, due to an asynchronous nature of communication between content scripts and devtools panel of this extension, its not helphul to use it in step by step debuging mode (cause injected content-script is also becomes paused).

Howether, you can delegate the comparison of your worker context objects to a main thread (or top context) for example:
```javascript
// worker
self.postMessage({type: 'compare', left, right});
```
```javascript
// main thread
workerInstance.addEventListener('message', (e) => {
  if (e.data.type === 'compare') {
    console.diff?.(e.data.left, e.data.right);
  }
});
```
Reminder: the code not for production
